### PR TITLE
PLNSRVCE-1585: flag throttled pipelineruns; filter overhead alert metric when throttled

### PR DESCRIPTION
**Pull request by gabemontero**
So recent bugs and inefficiencies in RHTAP have lead to stampedes of PipelineRuns that violated quota to the extreme.  This impacts pipeline controller performance not just because of shared cpu/mem, but because Tekton requeues Pod creation error events, backing up reconciliation, and leading to greater overhead.  As we already have explicit metrics for quota and node violations, we will shield the execution overhead metric, which aims to see if pipeline controller is overly falling behind when PipelineRuns are more evenly spread across the cluster.

See https://github.com/tektoncd/pipeline/blob/edddd7146df4af1597332105a0446ced92dd2673/pkg/reconciler/taskrun/taskrun.go#L677-L686 for the pipelinecontroller requeues (either 1 second !! or 1 minute)

@openshift-pipelines/pipelines-service FYI

### CODE REVIEW

**Review by Roming22 — APPROVED** [review]

File: collector/utils.go
Comment (enarha) [inline collector/utils.go:204]: Can we omit that? Doesn't the next line `_, previouslyLabelled := pr.Labels[THROTTLED_LABEL]` (:=) have the same effect?

Comment (gabemontero) [reply collector/utils.go:204]: yeah I should also instantiate a labels if it is nil ... I'll combine that with your suggestion - thanks

File: collector/utils.go
Comment (gabemontero) [reply collector/utils.go:183]: yep

**Comment by gabemontero** [comment]
updates pushed @Roming22 @enarha .... I need someone to reapply approvals before I get the merge button.  Or if one of you two already have the merge button, please submit that.

**Pull request merged**
